### DISCUSSION TITLE
I experience an awful performance of Avalonia demo Music Store app on iOS. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by kyurkchyan** [AvaloniaUI/Avalonia#13367] [Q&A] [answered]
Hey community 

It's been a week since I started learning Avalonia after a decade of Xamarin/Maui experience. So far, the development experience with Avalonia is fantastic. I've been following the official documentation to get started. I've built the `MusicStore` app. 

After that, I tried to adapt the `MusicStore` desktop app to the cross-platform variant so it could run on both desktop devices and iOS and Android. I've created a new project from Avalonia's Cross Platform project template and moved the code to that new project. The project is in [this GIT repository](https://github.com/kyurkchyan/MusicStore.Avalonia.Xplat). 

The first issue I faced was that the Music Store demo app is a multi-window app. iOS/Android doesn't support multiple windows. So, I had to change the navigation from dialog-based to view-based. I utilized ReactiveUI's routing system with `RoutedViewHost` for this. After some modifications and tinkering, I could run the same app on the desktop platform.

After all of those modifications, I could run the app successfully on Mac, Windows, and Android platforms. However, the performance is so bad it's not usable. Animations are not working when you try to search in the music store screen, it just hangs indefinitely. 

I was trying to find some information in the documentation suggesting that the iOS version is not usable, it's super experimental; don't try using it. But I couldn't. I saw a paragraph in [the release notes](https://docs.avaloniaui.net/docs/next/stay-up-to-date/whats-new#experimental-metal-support) suggesting experimental support for Metal that would improve performance, but I could not find any information about how to enable that.

So, what am I doing wrong? Or is this an expected behavior? It might be I am doing something awfully wrong, or it's just the fact that Avalonia.iOS is not production-ready yet, which also will be a totally acceptable for me.

**Comment by timunie** [comment]
first thing I'd look into: 

1. Try to use AOT if possible
2. only try release builds on the actual device, as debug will add significant overhead

I have no iOS so I cannot double-check on my own.

**Reply by kyurkchyan** [reply]
@timunie  
1. How can I enable AOT? 
2. I tried to debug both on the simulator and on a real device. I understand that there is a huge overhead when running in debug. But shouldn't we be able to test the app on the Simulator at least? Otherwise, the caveat of would be that the app will be completely untestable on iOS, which would stall the development process. 

Take a look at this video to see how the app looks on Simulator, even with release mode (without AOT though). 

https://github.com/AvaloniaUI/Avalonia/assets/3243139/73e479ac-2b32-41bf-a061-16cf316b87dc

In comparison, this is how it looks on Android

https://github.com/AvaloniaUI/Avalonia/assets/3243139/d5a1b64b-a449-48de-a47f-d79dab76020e

**Reply by timunie** [reply]
which Avalonia version are you using?

**Reply by kyurkchyan** [reply]
@timunie  It's `11.0.5`. The sample code is [here](https://github.com/kyurkchyan/MusicStore.Avalonia.Xplat)

**Comment by kyurkchyan** [comment, marked as answer]
@timunie  I've updated the version of Avalonia to `11.0.999-cibuild0035781-beta` and the issue was gone. Now, the app performs equally well on iOS. I would like to understand why, though. And perhaps also learn when will `11.0.999` be released to the public so we could boldly target iOS apps with Avalonia

**Reply by kyurkchyan** [reply]
It seems, the Metal-based is added in this later preview builds. Probably that's the reason

**Reply by timunie** [reply]
This commits will be in 11.1 available. I've no due date yet for you.
